code/modules/ftl/machinery/phase_cannon.dm

```diff
@@ -176,43 +181,54 @@
 						to_chat(user, "<span class='warning'>\The [WT] must be on to complete this task!</span>")
 					return
 				playsound(src.loc, W.usesound, 50, 1)
-				to_chat(user, "<span class='notice'>You start unwelding the cannons supports...</span>")
+				to_chat(user, "<span class='notice'>You start unwelding the supports of \the [src]...</span>")
 				if(do_after(user, 20*W.toolspeed, target = src))
 					if(!src || !WT.isOn()) return
 					to_chat(user, "<span class='notice'>You unweld the supports of \the [src].</span>")
 					anchored = FALSE
 					state = CONSTRUCTION_STATE2
 					W.add_fingerprint(user)
 
-			if(istype(W, /obj/item/stack/sheet/mineral/diamond)) //Contruct
-				var/obj/item/stack/sheet/mineral/diamond/D = W
-				if(D.use(5))
-					to_chat(user, "<span class='notice'>You install \the [D] as focusing lenses for \the [src].</span>")
-					state = CONSTRUCTION_STATE4
-				else
-					to_chat(user, "<span class='warning'>You require at least 5 [D] to install lenses for \the [src]!</span>")
+			if(istype(W, /obj/item/stack/sheet)) //Contruct
+				var/obj/item/stack/sheet/S = W
+				if(istype(S, /obj/item/stack/sheet/mineral/diamond)) //|| istype(S, /obj/item/stack/sheet/glass))
```

Comment:
is there a reason for this comment

Reply:
Originally I used to use that as one large if statement, but it became easier to split them, so done and forgot about.